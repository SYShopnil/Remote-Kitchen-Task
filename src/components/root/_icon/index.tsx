"use client";
import React from "react";

import { IoIosArrowDown } from "react-icons/io";
import { IIconStore, IconEnums } from "@src/types/root";
import { FaArrowAltCircleRight } from "react-icons/fa";
import { TbError404 } from "react-icons/tb";
import { MdDelete } from "react-icons/md";
import { MdModeEdit } from "react-icons/md";

import { Logo } from "./logo";

export const SIconStore = ({ iconName, fill }: IIconStore) => {
  switch (iconName) {
    case IconEnums.Logo: {
      return <Logo />;
    }
    case IconEnums.TbError404: {
      return <TbError404 fill={fill} />;
    }
    case IconEnums.FaArrowAltCircleRight: {
      return <FaArrowAltCircleRight fill={fill} />;
    }
    case IconEnums.MdDelete: {
      return <MdDelete fill={fill} />;
    }
    case IconEnums.MdModeEdit: {
      return <MdModeEdit fill={fill} />;
    }
    default: {
      return <IoIosArrowDown />;
    }
  }
};
