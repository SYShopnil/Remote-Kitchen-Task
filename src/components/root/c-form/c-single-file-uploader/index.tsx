"use client";
import React, { useState } from "react";
import singleFileUploaderStyle from "./singleFileUploaderStyle.module.css";

interface ICSingleFileUploader {
  filePassHandler: any;
  fileType: string;
}

export const CSingleFileUploader = ({
  filePassHandler,
  fileType,
}: ICSingleFileUploader) => {
  const [fileBase64, setFileBase64] = useState<any>("");
  const handleChange = (e: any) => {
    e.preventDefault();
    const file = e.target.files[0];

    if (file.name) {
      const { size } = file;
      const reader = new FileReader();

      // convert the data to base64
      reader.onloadend = () => {
        filePassHandler(reader.result, size);
        if (fileType == "img") {
          setFileBase64(reader.result);
        }
      };
      reader.readAsDataURL(file);
    }
  };

  return (
    <>
      <div className="mb-3">
        <input
          onChange={handleChange}
          placeholder=""
          type="file"
          className="form-control"
          required
        ></input>
      </div>
      {fileBase64 && (
        <img
          src={fileBase64}
          alt="upload_image"
          className={`${singleFileUploaderStyle.imageSize}`}
        />
      )}
    </>
  );
};
