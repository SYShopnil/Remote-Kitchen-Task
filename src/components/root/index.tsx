export { Button, CRedirectButton } from "./button";
export { SIconStore } from "./_icon/";
export { SIconWithMessage } from "./s-icon-with-message";
export { SProductCard } from "./s-product-card";
export { SLoading } from "./s-loading";
export { CIconWithHandlerButton } from "./c-icon-with-handler-button";
export { CForm } from "./c-form";
