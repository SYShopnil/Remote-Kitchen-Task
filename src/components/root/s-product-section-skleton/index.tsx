import React from "react";
import { RiSearchLine } from "react-icons/ri";

export const SProductSectionSkeleton = () => {
  const totalDemoCardSkeleton = Array(9).fill(0);
  return (
    <div>
      <div className="flex justify-evenly items-start space-x-2 pl-[2rem] ">
        {/* search bar */}
        <div className="flex-[1_1_80%] flex items-center justify-between bg-gray-200 rounded-md p-2">
          <div className="w-3/4 bg-gray-300 h-10 animate-pulse rounded-md"></div>
          <div className="w-1/4 bg-gray-300 h-10 animate-pulse rounded-md flex items-center justify-center">
            <RiSearchLine className="text-gray-400 h-6 w-6" />
          </div>
        </div>
        {/* button */}
        <div className="flex-[1_1_20%] ">
          <button className="bg-gray-300 text-gray-300 font-semibold py-2 px-4 rounded-md w-32 h-13 animate-pulse">
            Loading...
          </button>
        </div>
      </div>
      <div
        className={`grid grid-cols-12 gap-2  mt-[5rem] pl-[2rem] place-content-center`}
      >
        {totalDemoCardSkeleton.map((_, ind) => {
          return (
            <div
              role="status"
              key={ind}
              className="col-span-12  md:col-span-6  lg:col-span-4  max-w-sm p-4 border border-gray-200 rounded shadow animate-pulse md:p-6 dark:border-gray-700"
            >
              <div>
                <div className="flex items-center justify-center h-48 mb-4 bg-gray-300 rounded dark:bg-gray-700">
                  <svg
                    className="w-10 h-10 text-gray-200 dark:text-gray-600"
                    aria-hidden="true"
                    xmlns="http://www.w3.org/2000/svg"
                    fill="currentColor"
                    viewBox="0 0 16 20"
                  >
                    <path d="M14.066 0H7v5a2 2 0 0 1-2 2H0v11a1.97 1.97 0 0 0 1.934 2h12.132A1.97 1.97 0 0 0 16 18V2a1.97 1.97 0 0 0-1.934-2ZM10.5 6a1.5 1.5 0 1 1 0 2.999A1.5 1.5 0 0 1 10.5 6Zm2.221 10.515a1 1 0 0 1-.858.485h-8a1 1 0 0 1-.9-1.43L5.6 10.039a.978.978 0 0 1 .936-.57 1 1 0 0 1 .9.632l1.181 2.981.541-1a.945.945 0 0 1 .883-.522 1 1 0 0 1 .879.529l1.832 3.438a1 1 0 0 1-.031.988Z" />
                    <path d="M5 5V.13a2.96 2.96 0 0 0-1.293.749L.879 3.707A2.98 2.98 0 0 0 .13 5H5Z" />
                  </svg>
                </div>
                <div>
                  <div className="h-2.5 bg-gray-200 rounded-full dark:bg-gray-700 w-48 mb-4"></div>
                  <div className="h-2 bg-gray-200 rounded-full dark:bg-gray-700 mb-2.5"></div>
                  <div className="h-2 bg-gray-200 rounded-full dark:bg-gray-700 mb-2.5"></div>
                  <div className="h-2 bg-gray-200 rounded-full dark:bg-gray-700"></div>
                  <span className="sr-only">Loading...</span>
                </div>
              </div>
            </div>
          );
        })}
      </div>
    </div>
  );
};
