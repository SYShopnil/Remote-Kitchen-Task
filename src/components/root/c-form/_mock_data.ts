import { BtnColorSchema } from "@src/types/root";
import { FormElementEnum, IForm } from "@src/types/root/c-form";

export const mockProps: IForm = {
  buttonInfo: {
    btnText: "Demo One",
    colorSchema: BtnColorSchema.SolidBgGrayTextViolet,
    isArrow: false,
  },
  formFields: [
    {
      elementType: FormElementEnum.Input,
      label: "Food Name",
      space: 2,
      isRequired: true,
      name: "name",
      type: "input",
      placeholder: "Give Name",
    },
    {
      elementType: FormElementEnum.Input,
      label: "Food Price",
      space: 2,
      isRequired: true,
      name: "price",
      type: "input",
      placeholder: "Give Price",
    },
    {
      elementType: FormElementEnum.TextArea,
      label: "Description",
      space: 2,
      isRequired: true,
      name: "desc",
      type: "input",
      placeholder: "Give Description",
    },
    {
      elementType: FormElementEnum.Upload,
      label: "Upload Food Picture",
      space: 2,
      isRequired: true,
      name: "image",
      type: "input",
      placeholder: "Give image",
    },
  ],
  onSubmit: () => {},
  formTitle: "Test Form",
};
