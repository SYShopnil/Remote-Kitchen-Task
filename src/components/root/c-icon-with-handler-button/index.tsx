"use client";

import React from "react";
import { SIconStore } from "../_icon";
import { ICIconWithHandlerButton } from "@src/types/root/c-icon-with-handler-button";

export const CIconWithHandlerButton = ({
  icon: iconName,
  clickHandler,
}: ICIconWithHandlerButton) => {
  return (
    <button onClick={clickHandler} className={`bg-white`}>
      <SIconStore iconName={iconName} fill={"7F4D4F"} />
    </button>
  );
};
