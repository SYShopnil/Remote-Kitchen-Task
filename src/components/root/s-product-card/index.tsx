import { BtnColorSchema } from "@src/types/root";
import { CRedirectButton } from "../button";
import Image from "next/image";
import { EDataTestId } from "@src/types/common";
import { ISProductCard } from "@src/types/root/s-product-card";
import { CIconWithHandlerButtonContainer } from "@src/components/compound";
import { EMode } from "@src/types/compound/c-icon-with-handler-button-container";

export function SProductCard({
  desc,
  id: productId,
  image,
  name,
  price,
}: ISProductCard) {
  const currentSelectedProductDataForUpdatePhase = {
    name,
    desc,
    image,
    price,
    foodId: productId,
  };
  return (
    <div
      data-testid={EDataTestId.SProductCard}
      className={`h-auto  w-[100%] md:w-[18.75rem] text-center relative `}
    >
      <div
        className={`rounded-sm flex justify-center items-center w-[100%] lg:w-[18.75rem] min-h-[24rem] `}
      >
        <Image
          alt={name}
          src={image}
          placeholder="blur"
          blurDataURL="/assert/blur-demo-product.jpg"
          priority
          height={384}
          width={300}
        />
      </div>

      {/* below part container */}
      <div className={`h-[11.125rem] `}>
        <p className="text-[#777777] font-bold text-lg">{name}</p>
        <p className="text-[#79494B] font-bold text-[1rem] line-clamp-4">
          {price}$
        </p>
        <p className={`text-[#777777] font-bold text-sm line-clamp-2 mb-4`}>
          {desc}
        </p>
        <div className="flex justify-center items-center">
          <CRedirectButton
            btnText="View Details"
            colorSchema={BtnColorSchema.SolidBgWhiteTextGreen}
            isArrow={false}
            isOpenNewTab={false}
            redirectLink="/"
          />
        </div>
      </div>
      <div
        className={` absolute top-[3%] right-[10%] flex justify-evenly w-[5rem] bg-[#7F4D4F] p-2 rounded-md`}
      >
        <div>
          <CIconWithHandlerButtonContainer
            mode={EMode.UPDATE}
            foodId={productId}
            existFoodData={currentSelectedProductDataForUpdatePhase}
          />
        </div>

        <div>
          <CIconWithHandlerButtonContainer
            mode={EMode.DELETE}
            foodId={productId}
          />
        </div>
      </div>
    </div>
  );
}
