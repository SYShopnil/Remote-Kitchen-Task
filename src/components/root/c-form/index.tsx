"use client";

// External Imports
import { useState } from "react";
import Link from "next/link";
import { IForm, IFormValues } from "@src/types/root/c-form";
import { FormField } from "./c-form-field";
import { Button } from "../button";
import { EDataTestId } from "@src/types/common";

export const CForm = ({
  formTitle,
  formFields,
  buttonInfo,
  privacyInfo,
  onSubmit,
  initialValue = {},
}: IForm) => {
  const [formValues, setFormValues] = useState<IFormValues>(initialValue);
  const [formErrors, setFormErrors] = useState<IFormValues>(initialValue);
  console.log(formValues);
  const handleFormChange = (
    e: React.ChangeEvent<HTMLTextAreaElement | HTMLInputElement | any>
  ) => {
    const { name, value, type } = e.target;
    if (type === "checkbox" && e.target instanceof HTMLInputElement) {
      setFormValues({
        ...formValues,
        [name]: e.target.checked ? "true" : "",
      });
    } else {
      setFormValues({
        ...formValues,
        [name]: value,
      });
    }
  };
  const uploadHandler = (base64: any) => {
    setFormValues({
      ...formValues,
      image: base64,
    });
  };
  // submitHandler
  const submitHandler = (
    e: React.MouseEvent<HTMLElement> | React.FormEvent<HTMLFormElement>
  ) => {
    e.preventDefault();
    onSubmit(formValues);
  };

  return (
    <form
      className={`form grid gap-4 lg:grid-cols-2 p-3`}
      data-testid={EDataTestId.CForm}
    >
      {formTitle && (
        <div className="mb-[.5rem] md:mb-[.75rem] col-start-1 col-end-3">
          <h3 className="text-center text-[1.75rem] font-medium leading-[1.3] md:text-[3rem] md:font-bold md:leading-[1.1] capitalize text-secondary">
            {formTitle}
          </h3>
        </div>
      )}

      {formFields &&
        formFields.map((field, index) => {
          const halfWidth = "max-sm:col-start-1 max-sm:col-end-3";
          const fullWidth = "col-start-1 col-end-3";
          return (
            <div
              key={index}
              className={field.space === 0.5 ? halfWidth : fullWidth}
            >
              <FormField
                fieldProps={field}
                value={formValues[field.name] ? formValues[field.name] : ""}
                onChange={handleFormChange}
                formErrors={formErrors}
                uploadHandler={uploadHandler}
              />
            </div>
          );
        })}

      {privacyInfo && (
        <div className="col-start-1 col-end-3">
          <Link href={`${privacyInfo?.link}`} passHref>
            <a className="inline-flex focus:outline-0 focus:shadow-none focus-visible:outline-0 text-[.9375rem] text-secondary underline uppercase transition-colors duration-300 hover:text-secondaryLight">
              {privacyInfo?.linkText}
            </a>
          </Link>
        </div>
      )}

      {/* submit button */}
      {buttonInfo.btnText && (
        <div className="col-start-1 col-end-3">
          <Button {...buttonInfo} clickHandler={submitHandler} />
        </div>
      )}
    </form>
  );
};
