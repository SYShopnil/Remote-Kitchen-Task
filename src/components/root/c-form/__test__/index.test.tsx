import { render, screen } from "@testing-library/react";
import { EDataTestId } from "@src/types/common";
import { CForm } from "..";
import { FormElementEnum, IForm } from "@src/types/root/c-form";
import { BtnColorSchema } from "@src/types/root";
import { mockProps } from "../_mock_data";

describe("When Form  has rendered", () => {
  it("Expect it renders successfully", () => {
    const { getByTestId } = render(<CForm {...mockProps} />);
    const myElement = getByTestId(EDataTestId.CForm);
    // const myElement = screen.getByTestId(EDataTestId.CIconWithHandlerButtonContainer);
    expect(myElement).toMatchSnapshot();
  });
});
