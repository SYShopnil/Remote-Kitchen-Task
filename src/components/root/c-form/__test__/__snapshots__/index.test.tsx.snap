// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When Form  has rendered Expect it renders successfully 1`] = `
<form
  class="form grid gap-4 lg:grid-cols-2 p-3"
  data-testid="CForm"
>
  <div
    class="mb-[.5rem] md:mb-[.75rem] col-start-1 col-end-3"
  >
    <h3
      class="text-center text-[1.75rem] font-medium leading-[1.3] md:text-[3rem] md:font-bold md:leading-[1.1] capitalize text-secondary"
    >
      Test Form
    </h3>
  </div>
  <div
    class="col-start-1 col-end-3"
  >
    <div>
      <label
        class="text-base font-medium text-blackLight"
        for="name"
      >
        Food Name
      </label>
      <input
        autocomplete="Off"
        class="border-[1px] border-solid block w-full focus:outline-0 focus:shadow-none focus-visible:outline-0  placeholder:capitalize placeholder:text-blackLight text-blackLight bg-gray text-base rounded-lg p-[1rem_1.5rem] border-transparent"
        data-testid="name"
        id="name"
        name="name"
        placeholder="Give Name*"
        type="input"
        value=""
      />
    </div>
  </div>
  <div
    class="col-start-1 col-end-3"
  >
    <div>
      <label
        class="text-base font-medium text-blackLight"
        for="price"
      >
        Food Price
      </label>
      <input
        autocomplete="Off"
        class="border-[1px] border-solid block w-full focus:outline-0 focus:shadow-none focus-visible:outline-0  placeholder:capitalize placeholder:text-blackLight text-blackLight bg-gray text-base rounded-lg p-[1rem_1.5rem] border-transparent"
        data-testid="price"
        id="price"
        name="price"
        placeholder="Give Price*"
        type="input"
        value=""
      />
    </div>
  </div>
  <div
    class="col-start-1 col-end-3"
  >
    <div>
      <label
        class="text-base font-medium text-blackLight"
        for="desc"
      >
        Description
      </label>
      <textarea
        autocomplete="Off"
        class="border-[1px] border-solid block w-full focus:outline-0 focus:shadow-none focus-visible:outline-0  placeholder:capitalize placeholder:text-blackLight text-blackLight bg-gray text-base rounded-lg p-[1rem_1.5rem] border-transparent resize-none min-h-[130px]"
        data-testid="desc"
        id="desc"
        name="desc"
        placeholder="Give Description*"
      />
    </div>
  </div>
  <div
    class="col-start-1 col-end-3"
  >
    <div>
      <label
        class="text-base font-medium text-blackLight"
        for="image"
      >
        Upload Food Picture
      </label>
      <div>
        <div
          class="mb-3"
        >
          <input
            class="form-control"
            placeholder=""
            required=""
            type="file"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    class="col-start-1 col-end-3"
  >
    <span
      class="cursor-pointer"
    >
      <span
        class="flex justify-center items-center  "
      >
        <span
          class="text-base md:text-base_2  flex justify-center items-center px-[1rem] py-[0.5rem] rounded-[0.563rem] w-[100%] h-[100%] bg-[#EDEDED] hover:bg-[#7A67EE]  duration-[0.5s] text-[#7A67EE] hover:text-[#EDEDED]   border-[0.063rem] hover:border-[#EDEDED] border-[#7A67EE]   "
        >
          <span>
            Demo One
          </span>
        </span>
      </span>
    </span>
  </div>
</form>
`;
